test/blocks/gist/gist.test.js

```diff
@@ -6,10 +7,43 @@ document.body.innerHTML = await readFile({ path: './mocks/body.html' });
 const { default: init } = await import('../../../libs/blocks/gist/gist.js');
 
 describe('adobetv autoblock', () => {
-  it('creates iframe video block', async () => {
+  let createElementStub;
+  let originalCreateElement;
+
+  beforeEach(() => {
+    originalCreateElement = document.createElement;
+
+    createElementStub = sinon.stub(document, 'createElement').callsFake((tagName) => {
+      if (tagName === 'script') {
+        const scriptMock = originalCreateElement.call(document, 'script');
+        Object.defineProperty(scriptMock, 'src', {
+          set(url) {
+            const callbackMatch = url.match(/callback=([^&]+)/);
+            if (callbackMatch) {
+              const callbackName = callbackMatch[1];
+              Promise.resolve().then(() => {
+                if (window[callbackName]) {
```

Comment:
Why is this executed within a Promise resolve? should you rather use `clock.runAllAsync` if you awiat for a cb to execute?